docs/doxygen/Doxymain.md

```diff
@@ -0,0 +1,76 @@
+# Welcome to p4 Compiler's documentation!
+
+@htmlonly
+<div class="diagram-container">
+<!-- draw.io diagram -->
+<div class="mxgraph" style="" data-mxgraph="{&quot;highlight&quot;:&quot;#0000ff&quot;,&quot;lightbox&quot;:false,&quot;nav&quot;:true,&quot;edit&quot;:&quot;_blank&quot;,&quot;url&quot;:&quot;https://drive.google.com/uc?id=16X8Z_4-J8mHQPxuxm9HjWppPMTTXZdZ6&amp;export=download&quot;}"></div>
+<script type="text/javascript" src="https://viewer.diagrams.net/embed2.js?&fetch=https%3A%2F%2Fdrive.google.
```

Comment:
Ideally we should not rely on google drive here. Can we upload this to Github somehow?

Reply:
I tried but, couldn't find a way to embed it via GitHub without compromising on functionality.

Reply:
Draw.io supports sharing only from Google Drive.
I'll see if I can find a workaround.

Reply:
You should be able to save or export a .svg file from draw.io and save that anywhere you like, e.g. in this Github repo, yes?

Reply:
> You should be able to save or export a .svg file from draw.io and save that anywhere you like, e.g. in this Github repo, yes?

Indeed, I have included the diagram SVG and Draw.io file in this Pull Request.

Context - We want the SVG to be embedded with all functionality[links and dynamic toggle] while making sure that it is compatible with Doxygen output. 
Draw.io is supporting embedding option only with Google Drive.

Reply:
This is the only thing blocking us here right now. The problem is we do not have control over when draw.io goes down or things change there.

Reply:
Sorry that I didn't follow the design from the beginning so maybe this was already discussed. But I wonder whether https://github.com/mermaid-js/mermaid is an open-source alternative worth considering here?

Pros:
- Source can be checked in and maintained as plain text.
- Wide integrations including:
  - [Built-in GitHub rendering support](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams).
  - [VSCode plugin](https://marketplace.visualstudio.com/items?itemName=MermaidChart.vscode-mermaid-chart).
  - And [more](https://mermaid.js.org/ecosystem/integrations-community.html).

Cons:
- Might not be as capable as draw.io. So the resultant SVG might not be as visually appealing.

Reply:
What happens if you use an .svg file checked into Github as part of the page, instead of using draw.io?  Does something go wrong?

Reply:
> This is the only thing blocking us here right now. The problem is we do not have control over when draw.io goes down or things change there.

I'm looking into this and I think it will be possible after exporting the diagram as HTML and then keeping it in repo.(Still trying to implement the easter egg rest of the part is working)
 
I'll convert this PR to draft for the time!!

Reply:
> Sorry that I didn't follow the design from the beginning so maybe this was already discussed. But I wonder whether https://github.com/mermaid-js/mermaid is an open-source alternative worth considering here?

Thanks for the suggestion, it seems like a good alternative. However, Draw.io was already finalized for diagrams. 

> * Mermaid also probably doesn't support animation.

This is the main concern here, as current GitHub Hosted SVG in this PR can be easily embedded. But they don't support animation toggle.

Reply:
> What happens if you use an .svg file checked into Github as part of the page, instead of using draw.io? Does something go wrong?

The Easter egg toggle won't work!! Rest, there won't be any other change.

Reply:
Personally, I am fine giving up the Easter egg toggle if the docs page can rely only on files provided from one server, rather than requiring draw.io to be up and accessible as well.